src/gen_experiments/pdes.py

```diff
@@ -121,22 +121,75 @@ def kdv(t, u, dx, nx):
         "coeff_true": [
             {"u_11": -1, "u_1111": -1, "uu_1": -1},
         ],
-        "spatial_grid": np.arange(0, 10, 0.1),
     },
     "ks_periodic": {
         "rhsfunc": {"func": ks_periodic, "dimension": 1},
         "input_features": ["u"],
-        "time_args": [0.4, 100],
         "coeff_true": [
             {"u_11": -1, "u_1111": -1, "uu_1": -1},
         ],
-        "spatial_grid": np.linspace(0, 100, 1024),
     },
 }
 
 
+def data_prep(
+    seed,
+    group,
+    sim_params,
+    grid_params=None,
+    grid_vals=None,
+):
+    logger.info(f"Beginning Data Generation for {group}")
+    rhsfunc = pde_setup[group]["rhsfunc"]["func"]
+    dimension = pde_setup[group]["rhsfunc"]["dimension"]
+    spatial_grid = sim_params["spatial_grid"]
+    initial_condition = sim_params["init_cond"]
+    dt = sim_params["dt"]
+    spatial_args = [
+        (spatial_grid[-1] - spatial_grid[0]) / len(spatial_grid),
+        len(spatial_grid),
+    ]
+    if grid_params is not None:
+        start_time = process_time()
```

Comment:
the you can move calls to `process_time()` outside the conditional so we can remove redundant lines of code